src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/SourceGeneratorAttributeAnalyzerFix.cs

```diff
@@ -0,0 +1,109 @@
+                    context, document, node, diagnostic, LanguageNames.VisualBasic);
+
+                AddFix(
+                    string.Format(CultureInfo.CurrentCulture, CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_2, LanguageNames.CSharp, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, LanguageNames.CSharp, LanguageNames.VisualBasic);
+            }
+        }
+
+        private static void AddFix(string title, CodeFixContext context, Document document, SyntaxNode node, Diagnostic diagnostic, params string[] languageNames)
+        {
+            var codeAction = CodeAction.Create(
+                title,
+                (cancellationToken) => FixDocumentAsync(document, node, languageNames, cancellationToken),
+                equivalenceKey: nameof(SourceGeneratorAttributeAnalyzerFix));
+
+            context.RegisterCodeFix(codeAction, diagnostic);
+        }
+
+        public override FixAllProvider GetFixAllProvider() => null;
+
+        private static async Task<Document> FixDocumentAsync(Document document, SyntaxNode node, string[] languageNames, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+
+            SyntaxNode? generatorAttribute;
+
+            if (languageNames.Length == 1 && languageNames[0] == LanguageNames.CSharp)
+            {
+                // CSharp is the only language, which is the default paramterless 
+                // constructor for the Generator attribute
+                generatorAttribute = generator.Attribute(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute);
+            }
+            else
+            {
+                // For cases where the language is VB or VB and CSharp, add 
+                // an argument to signify that
+                var languageNamesFullName = typeof(LanguageNames).FullName;
+                var splitLanguageNames = languageNamesFullName.Split('.');
+
+                var baseLanguageNameExpression = generator.IdentifierName(splitLanguageNames[0]);
+                foreach (var identifier in splitLanguageNames.Skip(1))
+                {
+                    baseLanguageNameExpression = generator.MemberAccessExpression(baseLanguageNameExpression, identifier);
+                }
+
+                var arguments = new SyntaxNode[languageNames.Length];
+                for (var i = 0; i < languageNames.Length; i++)
+                {
+                    RoslynDebug.Assert(languageNames[i] == LanguageNames.CSharp || languageNames[i] == LanguageNames.VisualBasic);
+
+                    var language = languageNames[i] == LanguageNames.CSharp
+                        ? nameof(LanguageNames.CSharp)
+                        : nameof(LanguageNames.VisualBasic);
```

Comment:
I don't get how `language` is going to be different than `languageNames[i]`

Reply:
It definitely is not. This was prior to me adding the assert above, which is just a nice to have to formally state what languages are expected in this fix. Changing